bindings/azure/cosmosdb/cosmosdb.go

```diff
@@ -62,56 +59,57 @@ func (c *CosmosDB) Init(metadata bindings.Metadata) error {
 	c.partitionKey = m.PartitionKey
 
 	// Create the client; first, try authenticating with a master key, if present
-	var config *documentdb.Config
+	var client *azcosmos.Client
 	if m.MasterKey != "" {
-		config = documentdb.NewConfig(&documentdb.Key{
-			Key: m.MasterKey,
-		})
+		cred, keyErr := azcosmos.NewKeyCredential(m.MasterKey)
+		if keyErr != nil {
+			return keyErr
+		}
+		client, err = azcosmos.NewClientWithKey(m.URL, cred, nil)
+		if err != nil {
+			return err
+		}
 	} else {
 		// Fallback to using Azure AD
-		env, errB := azure.NewEnvironmentSettings("cosmosdb", metadata.Properties)
-		if errB != nil {
-			return errB
+		env, errEnv := azure.NewEnvironmentSettings("cosmosdb", metadata.Properties)
+		if errEnv != nil {
+			return errEnv
 		}
-		spt, errB := env.GetServicePrincipalToken()
-		if errB != nil {
-			return errB
+		token, errToken := env.GetTokenCredential()
+		if errToken != nil {
+			return errToken
 		}
-		config = documentdb.NewConfigWithServicePrincipal(spt)
-	}
-	// disable the identification hydrator (which autogenerates IDs if missing from the request)
-	// so we aren't forced to use a struct by the upstream SDK
-	// this allows us to provide the most flexibility in the request document sent to this binding
-	config.IdentificationHydrator = nil
-	config.WithAppIdentifier("dapr-" + logger.DaprVersion)
-
-	c.client = documentdb.New(m.URL, config)
-
-	// Retries initializing the client if a TooManyRequests error is encountered
-	err = retryOperation(func() (err error) {
-		collLink := fmt.Sprintf("dbs/%s/colls/%s/", m.Database, m.Collection)
-		coll, err := c.client.ReadCollection(collLink)
+		client, err = azcosmos.NewClient(m.URL, token, nil)
 		if err != nil {
-			if isTooManyRequestsError(err) {
-				return err
-			}
-			return backoff.Permanent(err)
-		} else if coll == nil || coll.Self == "" {
-			return backoff.Permanent(
-				fmt.Errorf("collection %s in database %s for CosmosDB state store not found. This must be created before Dapr uses it", m.Collection, m.Database),
-			)
+			return err
 		}
+	}
 
-		c.collection = coll.Self
-
-		return nil
-	}, func(err error, d time.Duration) {
-		c.logger.Warnf("CosmosDB binding initialization failed: %v; retrying in %s", err, d)
-	}, 5*time.Minute)
+	dbClient, err := client.NewDatabase(m.Database)
+	if err != nil {
+		return err
+	}
+	// Create a container client
+	dbContainer, err := dbClient.NewContainer(m.Collection)
+	if err != nil {
+		return err
+	}
+	if dbContainer.ID() == "" {
+		return backoff.Permanent(
+			fmt.Errorf("container is non-existent"),
+		)
+	}
+	c.client = dbContainer
+	containerRead, err := c.client.Read(context.Background(), nil)
```

Comment:
This needs a timeout because otherwise it could hang forever. For example:

```go
ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second)
containerRead, err := c.client.Read(ctx, nil)
cancel()
if err != nil {
    return err
}
```

Reply:
Should such a timeout be configurable via a new metadata property?

Reply:
No need given we don't have any other such metadata. But let's define these kinds of constants at the top of the file, not inline please!